application/Core/MORR/Data/IntermediateFormat/Json/JsonIntermediateFormatSerializer.cs

```diff
@@ -51,6 +51,16 @@ private void LinkAllQueues()
             }
         }
 
+        private void UnlinkOnClose()
+        {
+            var canClose = EventQueues.All(queue => queue.IsClosed);
+
+            if (canClose)
+            {
+                Close();
+            }
+        }
+
```

Comment:
Why is this necessary @insightmind? The class has a countdown event that automatically calls `Close` once all event queues have exited; this is specifically implemented in lines 93 - 96. This method here will not actually close the producer reliable if the event queues aren't already closed once this is called, which is not really guaranteed.

Reply:
Sorry this is was temporary code while I was trying to figure out why it did not work. Will remove it asap.